kroxylicious-annotations/src/main/java/io/kroxylicious/proxy/tag/CompletesOnThread.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.tag;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.function.Function;
+
+/**
+ * Annotation to document the thread on which an
+ * asynchronous result callback, such as might be passed to
+ * {@link java.util.concurrent.CompletionStage#thenApply(Function)}, will be executed.
+ * The intention is that this annotation is applied to methods which return CompletionStage,
+ * parameters of type CompletionStage, and so on.
+ */
+@Documented
+@Target({ ElementType.TYPE_USE, ElementType.PARAMETER,
+        ElementType.METHOD,
+        ElementType.FIELD,
+        ElementType.LOCAL_VARIABLE,
+        ElementType.RECORD_COMPONENT })
+@Retention(RetentionPolicy.SOURCE)
+public @interface CompletesOnThread {
+    /**
+     * A description of the thread on which the {@link java.util.concurrent.CompletableFuture},
+     * {@link java.util.concurrent.CompletionStage} or other asynchronous result
+     * is completed. Use "*" if the asynchronous result can complete on any thread
+     * (for example on usage in an interface which doesn't define a policy for implementors)
+     * @return
+     */
+    String value();
```

Comment:
I kinda feel like we should have a constant for `filter thread`. It also leaves me wondering if value should actually be an Enum 
```
enum ThreadContexts {
FILTER_THREAD,
MANAGED_POOL, // Aka a netty eventLoop
SHARED_EXECUTOR,
DEDICATED_THREAD,
ANY_THREAD
}
```

The names could be anything but I do wonder if we should be trying to convey more context about execution.

Reply:
No real objection to having constants. But I think an enum would be unnecessarily limiting. If a new thread gets used in any module that's using the annotation then we'd have to add a new enum constant. That would guarantee global uniqueness, but it seems a bit heavy weight to me.

Reply:
I was thinking enum to try and drive a bit of consistency in contract, to get people thinking about the category rather than the specifics. So yes it is restrictive but thats deliberate.

So to cover the case of 
```java
@CompletesOnThread("T") CompletionStage<List<T>> join(List<? extends @CompletesOnThread("T") CompletionStage<T>> stages) 
```
it might become 
```java
@CompletesOnThread(INPUT_THREAD) CompletionStage<List<T>> join(List<? extends @CompletesOnThread(ANY_THREAD) CompletionStage<T>> stages) 
```
As that is the contract you described.